generators/php/model/src/union/UnionGenerator.ts

```diff
@@ -0,0 +1,1329 @@
+            return_: php.Type.reference(this.classReference),
+            body,
+            static_: true
+        });
+    }
+
+    private staticConstructor(variant: SingleUnionType): php.Method {
+        return php.method({
+            name: this.context.getPropertyName(variant.discriminantValue.name),
+            access: "public",
+            parameters: this.getStaticConstructorParameters(variant),
+            return_: php.Type.reference(this.classReference),
+            body: this.getStaticConstructorBody(variant),
+            static_: true
+        });
+    }
+
+    private getStaticConstructorParameters(variant: SingleUnionType): php.Parameter[] {
+        const parameters: php.Parameter[] = [];
+
+        for (const property of this.unionTypeDeclaration.baseProperties) {
+            parameters.push(
+                php.parameter({
+                    name: this.context.getPropertyName(property.name.name),
+                    type: this.context.phpTypeMapper.convert({ reference: property.valueType })
+                })
+            );
+        }
+
+        switch (variant.shape.propertiesType) {
+            case "samePropertiesAsObject":
+            case "singleProperty":
+                parameters.push(
+                    php.parameter({
+                        name: this.context.getPropertyName(variant.discriminantValue.name),
+                        type: this.getReturnType(variant)
+                    })
+                );
+                break;
+            case "noProperties":
+            default:
+        }
+
+        return parameters;
+    }
+
+    private getStaticConstructorBody(variant: SingleUnionType) {
+        return php.codeblock((writer) => {
+            const constructorArgs: php.Map.Entry[] = [];
+
+            for (const property of this.unionTypeDeclaration.baseProperties) {
+                constructorArgs.push({
+                    key: php.codeblock(`'${property.name.wireValue}'`),
+                    value: php.codeblock(this.context.getVariableName(property.name.name))
+                });
+            }
+
+            constructorArgs.push({
+                key: php.codeblock(`'${this.unionTypeDeclaration.discriminant.wireValue}'`),
+                value: php.codeblock(`'${variant.discriminantValue.wireValue}'`)
+            });
+
+            switch (variant.shape.propertiesType) {
+                case "samePropertiesAsObject":
+                case "singleProperty":
+                    constructorArgs.push({
+                        key: php.codeblock("'value'"),
+                        value: php.codeblock(this.context.getVariableName(variant.discriminantValue.name))
+                    });
+                    break;
+                case "noProperties":
+                default:
+                    constructorArgs.push({
+                        key: php.codeblock("'value'"),
+                        value: php.codeblock("null")
+                    });
+            }
```

Comment:
Same thing as above with `assertNever` in the `default` case - we can move this logic in `noProperties` on its own.
```suggestion
            switch (variant.shape.propertiesType) {
                case "samePropertiesAsObject":
                case "singleProperty":
                    constructorArgs.push({
                        key: php.codeblock("'value'"),
                        value: php.codeblock(this.context.getVariableName(variant.discriminantValue.name))
                    });
                    break;
                case "noProperties":
                    constructorArgs.push({
                        key: php.codeblock("'value'"),
                        value: php.codeblock("null")
                    });
                default:
                    assertNever(variant.shape);
            }
```